lib/hooks/index.js

```diff
@@ -97,25 +67,51 @@ function findModuleVersion(request, parent, load) {
     if (mainScriptDir.indexOf(paths[i]) === 0) {
       // This is unsafe on windows if request contains a namespace
       // TODO: convert path separators in request
-      var pjson = load(path.join(paths[i], request, 'package.json'));
-      return pjson.version;
+      return path.join(paths[i], request);
     }
   }
   return null;
 }
 
+/**
+ * Determines the version of the module located at `modulePath`.
+ *
+ * @param {?string} modulePath The absolute path to the root directory of the
+ *    module being loaded. This may be null if we are loading an internal module
+ *    such as http.
+ */
+function findModuleVersion(modulePath, load) {
+  return modulePath ? load(path.join(modulePath, 'package.json')).version :
```

Comment:
Alternatively, you can make sure that `Module._load` behaves predictably in the case the file doesn't exist (e.g. doesn't throw or crash), and check the return value. I would expect `Module._load` already needs to do a file existence check; it would good not to repeat it ourselves if at all possible.